### PR TITLE
nitial commit

### PR DESCRIPTION
**Pull request by lanre-ade**
Not ready

### CODE REVIEW

File: api/models/author.py [hunk 1]
```diff
@@ -60,15 +60,16 @@ def add_pending(self, friend):
...
     def add_friend(self, friend):
         """Create a friend from an author model"""
-        friend = self._get_cached_author(friend)
-        if not self.friends.filter(id=friend.id).exists():
-            self.friends.add(friend)
+        friend_cached = self._get_cached_author(friend)
+        if not self.friends.filter(id=friend_cached.id).exists():
+            self.friends.add(friend_cached)
             self.save()
+            friend.add_friend(self)
```
Comment (derekdowling) [inline api/models/author.py:71]: This line doesn't make sense, hopefully its a wip.

Comment (mraypold) [inline api/models/author.py:71]: This will potentially cause type errors. Friend might not be an author model.

Comment (derekdowling) [inline api/models/author.py:71]: I would say I am in agreement that we shouldn't be dealing with adding friends to other objects inside this instance. That being said, add_friend should still be a model call. Whatever view designates that two authors should still become friends should be calling this on each respective object.

Comment (mraypold) [inline api/models/author.py:71]: I think using the ModifyRelationsMixin is still the best approach. Alternatively, we can also use signals to check every time a friends/followers attribute has been modified and then update the corresponding local author.

**Comment by xkb** [comment]
Is this PR still being worked on?

**Comment by derekdowling** [comment]
I don't think so. @mraypold has a different solution for PR.

**Pull request closed without merging**